services/compute/virtual-machines/taxonomy.md

```diff
@@ -0,0 +1,47 @@
+|	CCC-030111	|	Isolated Secure Environments	|	Providing an isolated "enclave" within a virtual machine for processing encrypted and/or sensitive data, with support for custom key management infrastructure.	|
+|	CCC-030112	|	Nested Virtualization	|	Allowing the creation of virtual machines within virtual machines.	|
+|	CCC-030113	|	Container Support	|	Offering support for running containers within virtual machine instances for containerized applications.	|
+|	CCC-030114	|	Instance Metadata	|	Providing metadata about virtual machine instances for configuration and management purposes.	|
+|	CCC-030115	|	Instance Lifecycle Events	|	Offering features for managing the lifecycle and state of virtual machine instances, including starting, stopping, pausing, and restarting instances as needed.	|
+|	CCC-030116	|	Instance Snapshots	|	Creation of snapshots of virtual machine instances to capture and preserve state and data for backup and cloning purposes. 	|
+|	CCC-030117	|	Instance Templates	|	Offering templates for provisioning virtual machine instances with pre-configured images, predefined configurations and settings and/or bootstrap scripts.	|
+|	CCC-030118	|	Instance Preemptibility	|	Providing the option for using preemptible virtual machine (spot) instances at a lower cost for non-critical or fault-tolerant workloads that may be terminated by the cloud provider after a notice period.	|
+|	CCC-030119	|	Instance Affinity/Anti-affinity	|	Enabling control over the location of virtual machine instances to ensure or prevent co-location on the same physical hardware.	|
+|	CCC-030120	|	Instance Health Checks	|	Providing features for performing health checks on virtual machine instances and automatically replacing or repairing unhealthy instances.	|
+|	CCC-030121	|	Instance Remote Access	|	Offering remote access to virtual machine instances through methods such as SSH or RDP for troubleshooting, debugging, and maintenance purposes.	|
+|	CCC-030122	|	Instance Live Migration	|	The ability to perform live migration of virtual machine instances between physical hosts for maintenance or load balancing purposes without downtime.	|
+|	CCC-030123	|	Instance Remote Configuration	|	Providing tools for remotely configuring virtual machine instances, including deployment automation and configuration management frameworks.	|
+|	CCC-030124	|	Instance Resource Tagging	|	Enabling tagging of virtual machine instances with metadata for organization, management, and cost allocation purposes.	|
+|	CCC-030125	|	Instance Resource Monitoring	|	Providing tools for monitoring resource utilization and performance metrics for virtual machine instances, including CPU usage, memory usage, disk I/O, and network traffic.	|
+|	CCC-030126	|	Custom Images	|	Allows users to create and manage their own customized virtual machine images.	|
```

Comment:
Should there be another feature to cover bootstrap scripts that is pre configured to run on start

Reply:
That's about right for an instance template - you're pre-determining instance type, specs and networking among other things. I've given bootstrap scripts their own point now.